File: SkillSurgeUI/src/app/core/models/interfaces/Category.ts
export interface Category {
    id: string;
    name: string;
    description?: string;
    isDeleted: boolean;
    parentCategoryId?: string,
    createdAt?: Date;
    updatedAt?: Date;
    activeSubCategories?: Category[];
    deletedSubCategories?: Category[];
    activeSubCategoriesCount?: number;
    deletedSubCategoriesCount?: number;
    isActiveExpanded?: boolean;
    isDeletedExpanded?: boolean;
}

File: Domain/Entities/Category.cs
namespace Domain.Entities;

public class Category : AuditableEntity
{
    public Guid Id { get; set; }
    public string Name { get; set; } = default!;
    public string? Description { get; set; }
    public Guid? ParentCategoryId { get; set; }
    public bool IsDeleted { get; set; }
}